classes/naas_client.php

```diff
@@ -55,57 +60,112 @@ public function __construct($config) {
      * @return proxy_http_response
      */
     public function request_raw($protocol, $service, $data = null, $params = null) {
+        global $CFG;
+
         $url = $this->config->naas_endpoint.$service;
         if ($params != null) {
             // Remove indices from query params.
             $query = preg_replace('/\%5B\d+\%5D/', '', http_build_query($params));
             $url .= "?".$query;
         }
-        $ch = curl_init();
-        $headers = [];
-        curl_setopt($ch, CURLOPT_URL, $url);
-        curl_setopt($ch, CURLOPT_USERPWD, $this->config->naas_username.":" . $this->config->naas_password);
-        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
-        if (property_exists($this->config, "proxyhost") && !empty($this->config->proxyhost)) {
-            curl_setopt($ch, CURLOPT_PROXY, $this->config->proxyhost);
-            curl_setopt($ch, CURLOPT_NOPROXY, $this->config->proxybypass);
-            curl_setopt($ch, CURLOPT_PROXYPORT, $this->config->proxyport);
+
+        // Log the request and config.
+        if ($this->debug) {
+            debugging("NAAS: Connecting to " . $url, DEBUG_DEVELOPER);
+            debugging("NAAS: Configuration: " . var_export($this->config, true), DEBUG_DEVELOPER);
         }
+
+        // Create curl with the correct Moodle class.
+        $curl = new \curl();
+        $headers = [];
+        $options = [
+            'CURLOPT_RETURNTRANSFER' => true,
+            'CURLOPT_USERPWD' => $this->config->naas_username . ":" . $this->config->naas_password,
+            'CURLOPT_CONNECTTIMEOUT' => 10, // Connection timeout.
+            // Explicitly disable proxy usage for all connections.
+            'CURLOPT_PROXY' => '',
```

Comment:
I do not understand.
The point using the Moodle CURL is to have the Moodle proxy config automatically applied.
Why disable it explicitly ?